src/analyze/analyze-chid.c

```diff
@@ -223,6 +247,87 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(sd_device *drm_card, char16_t **ret_panel) {
+        assert(drm_card);
+        assert(ret_panel);
+        const char *edid_content;
+        size_t edid_size;
+        int r;
+
+        r = sd_device_get_sysattr_value_with_size(drm_card, "edid", &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+        if (edid_size == 0)
+                return -ENODEV;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EBADMSG;
+
+        _cleanup_free_ char16_t *panel_id = new0(char16_t, 8);
+        if (!panel_id)
+                return log_oom();
+
+        if (!edid_get_panel_id(&header, panel_id))
+                return -EBADMSG;
+
+        *ret_panel = TAKE_PTR(panel_id);
+        return 0;
+}
+
+static int edid_search(char16_t **ret_panel) {
+        assert(ret_panel);
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        _cleanup_strv_free_ char **drm_paths = new0(char *, 1);
+        size_t n = 0;
+        int r;
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_allow_uninitialized(e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_subsystem(e, "drm", true);
+        if (r < 0)
+                return r;
+
+        FOREACH_DEVICE(e, d) {
+                const char *drm_path;
+                r = sd_device_get_syspath(d, &drm_path);
+                if (r < 0)
+                        return log_device_error_errno(d, r, "Couldn't get syspath from device: %m");
+
+                r = edid_parse(d, ret_panel);
+                if (r == -ENOENT)
+                        continue;
+                if (r < 0) {
+                        log_debug("Skipping DRM device with invalid EDID: %s", drm_path);
```

Comment:
we usually suffix our error messages with `…, ignoring: %m"` if we ignore an error, like here.